pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -1072,7 +1109,11 @@ impl LayerInner {
             LayerResidenceEventReason::ResidenceChange,
         );
 
-        let res = match capture_mtime_and_remove(&self.path) {
+        let local_path = self
+            .local_path()
+            .map_err(|_| EvictionCancelled::TimelineGone)?;
```

Comment:
But now we would be upgrading the timeline all over? It must already be upgraded to get to here because otherwise we would be running without a span.